### ISSUE TITLE
Supported job frameworks with TAS

### ISSUE THREAD

**Issue by kerthcet** [kubernetes-sigs/kueue#3886]
**What would you like to be cleaned**:

What's the current supported job frameworks with TAS, batch job etc., it will be great to list them under https://kueue.sigs.k8s.io/docs/concepts/topology_aware_scheduling/.

**Why is this needed**:

**Comment by mimowo** [comment]
Good idea! We could also elaborate a bit more on how to use it for Pod Groups with rank-based ordering.
cc @mwysokin @PBundyra

**Comment by kerthcet** [comment]
Can you give a short summary about the job framework support right now? @mimowo

**Comment by mimowo** [comment]
Sure, the baseline TAS is supported in all Jobs which call `jobframework.PodSetTopologyRequest`: https://github.com/search?q=repo%3Akubernetes-sigs%2Fkueue+jobframework.PodSetTopologyRequest&type=code. AFAIK this are all workloads: Job, JobSet, Kubeflow, Ray, Pod Groups.

The rank-based ordering is an enhancement on top of that available for Jobs which call this function with the label indexing the pods, such as [here](https://github.com/kubernetes-sigs/kueue/blob/f98a2f99d49e0ffcd8c8c7515e1328ffd398f1c9/pkg/controller/jobs/kubeflow/kubeflowjob/kubeflowjob_controller.go#L108) for Kubeflow. To make rank-based ordering for PodGroups we introduced a dedicated label called "pod-group-pod-index". Moreover, for more flexibility we have "pod-group-pod-index-label" which indicates another third-party label which might be indexing the pods.

**Comment by mimowo** [comment]
cc @andrewsykim are the pods created by RayJob or RayCluster indexed by a label or annotation?

**Comment by kerthcet** [comment]
Thanks, we mostly want to try this with Kubeflow jobs.

**Comment by andrewsykim** [comment]
> cc @andrewsykim are the pods created by RayJob or RayCluster indexed by a label or annotation?

No, is this required for TAS?

**Comment by PBundyra** [comment]
> 
> No, is this required for TAS?
> 

It is not required to use TAS in general. However if pods are indexed, TAS will place neighbouring pods next to each other if properly configured

**Comment by tenzen-y** [comment]
This is described in https://kueue.sigs.k8s.io/docs/overview/#job-integrated-features

/close
